ray-operator/controllers/ray/rayjob_controller.go

```diff
@@ -617,6 +655,31 @@ func (r *RayJobReconciler) deleteClusterResources(ctx context.Context, rayJobIns
 	return isClusterDeleted, nil
 }
 
+func (r *RayJobReconciler) scaleWorkerReplicasToZero(ctx context.Context, rayJobInstance *rayv1.RayJob) (bool, error) {
```

Comment:
Typically, a K8s controller should only write to the CR status and treat the CR spec as read-only, but implementing this feature without writing to the CR spec is challenging for us.

Perhaps a compromise solution is to add a new field to the RayCluster CRD (e.g., `suspendWorkers: bool`), where the RayJob controller only sets this field to true, and the RayCluster is responsible for deleting all Ray worker Pods.

This way, the RayJob controller doesn't need to modify replicas and minReplicas, which can also be modified by the Ray Autoscaler or users. Allowing multiple stakeholders to modify a field is typically the root cause of KubeRay's instability issues.

Reply:
Controllers writing to spec is not necessarily bad, but I see what you mean. I feel that updating replcias, minReplicas and maxReplicas for ephemeral RayCluster is actually fine.

Will think about this more and get back to you.

Reply:
The main concern is that multiple personas can modify these fields, such as users, the Autoscaler, and the RayJob controller.

Reply:
> Perhaps a compromise solution is to add a new field to the RayCluster CRD (e.g., suspendWorkers: bool), where the RayJob controller only sets this field to true, and the RayCluster is responsible for deleting all Ray worker Pods.

@kevin85421 how about a `suspend` field per worker group in WorkerGroupSpec? This allows for granularity of suspension per worker, and from RayJob we can just set `suspend: true` for all worker groups

Reply:
Here's the draft PR https://github.com/ray-project/kuberay/pull/2663 

Let me know what you think, I will clean up the PR and add tests if the API looks good to you

Reply:
The API looks good to me.